src/odinapi/utils/datamodel.py

```diff
@@ -0,0 +1,432 @@
+    dimension = attr.ib(type=Dimension)
+    l2type = attr.ib(type=L2Type)
+
+    def get_description(self, product: str) -> str:
+        if self.name == "Profile":
+            return (
+                "Retrieved temperature profile."
+                if is_temperature(product)
+                else "Retrieved volume mixing ratio."
+            )
+        return self.description
+
+    def get_units(self, product: str) -> Units:
+        if self.name == "AVK":
+            return Units.koverk if is_temperature(product) else Units.poverp
+        elif self.units != Units.product:
+            return self.units
+        else:
+            return (
+                Units.temperature if is_temperature(product)
+                else Units.unitless
+            )
+
+
+@attr.s
+class L2File:
+    parameters = attr.ib(type=List[Parameter])
+
+
+@attr.s
+class L2anc:
+    LST = attr.ib(type=float)
+    Orbit = attr.ib(type=int)
+    SZA1D = attr.ib(type=float)
+    SZA = attr.ib(type=List[float])
+    Theta = attr.ib(type=List[float])
+
+
+@attr.s
+class L2:
+    InvMode = attr.ib(type=str)
+    ScanID = attr.ib(type=int)
+    Time = attr.ib(type=dt.datetime)
+    Lat1D = attr.ib(type=float)
+    Lon1D = attr.ib(type=float)
+    Quality = attr.ib(type=float)
+    Altitude = attr.ib(type=List[float])
+    Pressure = attr.ib(type=List[float])
+    Profile = attr.ib(type=List[float])
+    Latitude = attr.ib(type=List[float])
+    Longitude = attr.ib(type=List[float])
+    Temperature = attr.ib(type=List[float])
+    ErrorTotal = attr.ib(type=List[float])
+    ErrorNoise = attr.ib(type=List[float])
+    MeasResponse = attr.ib(type=List[float])
+    Apriori = attr.ib(type=List[float])
+    VMR = attr.ib(type=List[float])
+    AVK = attr.ib(type=List[List[float]])
+
+
+@attr.s
+class L2i:
+    GenerationTime = attr.ib(type=dt.datetime)
+    Residual = attr.ib(type=float)
+    MinLmFactor = attr.ib(type=float)
+    FreqMode = attr.ib(type=int)
+
+    @property
+    def filter(self) -> Filter:
+        return Filter(
+            residual=1.5,
+            minlmfactor=10. if self.FreqMode in [8., 13., 19.] else 2.
+        )
+
+    def isvalid(self) -> bool:
+        return (
+            self.Residual <= self.filter.residual
+            and self.MinLmFactor <= self.filter.minlmfactor
+        )
+
+
+@attr.s
+class L2Full:
+    l2i = attr.ib(type=L2i)
+    l2anc = attr.ib(type=L2anc)
+    l2 = attr.ib(type=L2)
+
+    def get_data(self, parameter: Parameter):
+        if parameter.l2type == L2Type.l2i:
+            return getattr(self.l2i, parameter.name)
+        elif parameter.l2type == L2Type.l2anc:
+            return getattr(self.l2anc, parameter.name)
+        return getattr(self.l2, parameter.name)
+
+
+def get_file_header_data(
+        freqmode: int,
+        invmode: str,
+        product: str,
+        time_coverage_start: dt.datetime,
+        time_coverage_end: dt.datetime
+) -> Dict[str, str]:
+    header_data = {
```

Comment:
why is everything attrs but not this one?

Reply:
because I had this stuff from before, and this part is not very complicated, so I though it was good enough.